src/main/java/org/folio/inventory/dataimport/handlers/actions/AbstractInstanceEventHandler.java

```diff
@@ -58,4 +91,142 @@ protected org.folio.Instance defaultMapRecordToInstance(DataImportEventPayload d
+          LOGGER.warn(msg);
+          deleteInstance(instance.getId(), payload.getJobExecutionId(), instanceCollection);
+          promise.fail(msg);
+        }
+      });
+    return promise.future();
+  }
+
+  protected Future<Instance> putRecordInSrsAndHandleResponse(DataImportEventPayload payload, Record srcRecord,
+                                                             Instance instance, String matchedId) {
+    Promise<Instance> promise = Promise.promise();
+    getSourceStorageRecordsClient(payload).putSourceStorageRecordsGenerationById(matchedId ,srcRecord)
+      .onComplete(ar -> {
+        var result = ar.result();
+        if (ar.succeeded() && result.statusCode() == HttpStatus.HTTP_OK.toInt()) {
+          payload.getContext().put(EntityType.MARC_BIBLIOGRAPHIC.value(),
+            Json.encode(encodeParsedRecordContent(result.bodyAsJson(Record.class))));
+          LOGGER.info("Update MARC record in SRS with id: '{}', instanceId: '{}', from tenant: {}, jobExecutionId: {}",
+            srcRecord.getId(), instance.getId(), payload.getTenant(), payload.getJobExecutionId());
+          promise.complete(instance);
+        } else {
+          String msg = format("Failed to update MARC record in SRS, instanceId: '%s', jobExecutionId: '%s', status code: %s, Record: %s",
+            instance.getId(), payload.getJobExecutionId(), result != null ? result.statusCode() : "", result != null ? result.bodyAsString() : "");
+          LOGGER.warn(msg);
+          promise.fail(msg);
+        }
+      });
+    return promise.future();
+  }
+
+  protected Future<Instance> executeFieldsManipulation(Instance instance, Record srcRecord) {
+    AdditionalFieldsUtil.fill001FieldInMarcRecord(srcRecord, instance.getHrid());
+    if (StringUtils.isBlank(srcRecord.getMatchedId())) {
+      srcRecord.setMatchedId(srcRecord.getId());
+    }
+    setExternalIds(srcRecord, instance);
+    return AdditionalFieldsUtil.addFieldToMarcRecord(srcRecord, TAG_999, 'i', instance.getId())
+      ? Future.succeededFuture(instance)
+      : Future.failedFuture(format("Failed to add instance id '%s' to record with id '%s'", instance.getId(), srcRecord.getId()));
+  }
+
+  /**
+   * Adds specified externalId and externalHrid to record and additional custom field with externalId to parsed record.
+   *
+   * @param srcRecord   record to update
+   * @param instance externalEntity in Json
+   */
+  protected void setExternalIds(Record srcRecord, Instance instance) {
+    if (srcRecord.getExternalIdsHolder() == null) {
+      srcRecord.setExternalIdsHolder(new ExternalIdsHolder());
+    }
+    String externalId = srcRecord.getExternalIdsHolder().getInstanceId();
+    String externalHrid = srcRecord.getExternalIdsHolder().getInstanceHrid();
+    if (isNotEmpty(externalId) || isNotEmpty(externalHrid)) {
+      if (AdditionalFieldsUtil.isFieldsFillingNeeded(srcRecord, instance)) {
+        executeHrIdManipulation(srcRecord, instance.getJsonForStorage());
```

Comment:
logic was replaced as to be identical in source storage. [link](https://github.com/folio-org/mod-source-record-storage/blob/46f0abd66ad9f24c4c016834728b5cffda8049e6/mod-source-record-storage-server/src/main/java/org/folio/services/handlers/AbstractPostProcessingEventHandler.java#L276)